plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/condition/TableHasRows.java

```diff
@@ -10,30 +10,93 @@
  ******************************************************************************/ 
 package org.jboss.reddeer.swt.condition;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.hamcrest.Matcher;
 import org.jboss.reddeer.common.condition.AbstractWaitCondition;
 import org.jboss.reddeer.swt.api.Table;
+import org.jboss.reddeer.swt.api.TableItem;
 
 /**
  * Condition is met when table has rows.
  * 
- * @author Rastislav Wagner
+ * @author Rastislav Wagner, Josef Kopriva
  */
 public class TableHasRows extends AbstractWaitCondition {
 	private final Table table;
-
+	private final Matcher<?>[] matchers;
+	private final Integer column;
+	
 	/**
 	 * Constructs TableHasRows wait condition.
 	 * Condition is met when table has any rows.
 	 * 
 	 * @param table table which should contain rows to the condition be met
 	 */
 	public TableHasRows(Table table) {
+		this(table,null,null);
+	}
+	
+	
+	/**
+	 * Constructs TableHasRow wait condition.
+	 * Condition is met when table has specific row matched with matcher.
+	 * 
+	 * @param table table which should contain row to the condition be met
+	 * @param matchers matchers - e.g. Regexmatcher, custom matcher...
+	 */
+	public TableHasRows(Table table, Matcher<?>... matchers) {
+		this(table,null,matchers);
+	}
+	
+	/**
+	 * Constructs TableHasRow wait condition.
+	 * Condition is met when table has specific row and value in specific column matched with matcher.
+	 * 
+	 * @param table table which should contain row to the condition be met
+	 * @param column column in which should be used matchers
+	 * @param matchers matchers - e.g. Regexmatcher, custom matcher...
+	 */
+	public TableHasRows(Table table, Integer column, Matcher<?>... matchers) {
 		this.table = table;
+		this.matchers = matchers!=null?createMatchers(matchers):null;
+		this.column = column!=null?column.intValue():null;
+	}
+	
+	private static Matcher<?>[] createMatchers(Matcher<?>... matchers) {
+		List<Matcher<?>> list= new ArrayList<Matcher<?>>();		
+		list.addAll(Arrays.asList(matchers));
+		return list.toArray(new Matcher[list.size()]);
+	}
+	
+	private boolean search(int begin, int end){
+		List<TableItem> matchedItems = new ArrayList<TableItem>();
+		
+		for (TableItem item : table.getItems()){
+			for(int i=begin; i<end;i++){
+				int index = 0;
+				while (index < matchers.length && matchers[index].matches(item.getText(i))){
+					index++;
+				}
+				if (index == matchers.length){
+					matchedItems.add(item);
+				}
+			}
+		}
+		return !matchedItems.isEmpty();
 	}
 
 	@Override
 	public boolean test() {
-		return table.rowCount() > 0;
+		if(this.table!=null&&this.matchers==null&&this.column==null){
```

Comment:
Could you add some spaces, please? I recommend to use the auto formatting in Eclipse.